Sources/TuistCore/GraphTraverser/GraphTraversing.swift

```diff
@@ -107,6 +107,12 @@ public protocol GraphTraversing {
     ///   - name: Target name.
     func appExtensionDependencies(path: AbsolutePath, name: String) -> Set<GraphTarget>
 
+    /// Given a project directory and a target name, it returns all the dependencies that are extensions with ther condition
+    /// - Parameters:
+    ///   - path: Path to the directory that contains the project.
+    ///   - name: Target name.
+    func appExtensionDependenciesWithConditions(path: AbsolutePath, name: String) -> [(GraphTarget, PlatformCondition?)]
+
```

Comment:
We'd also need to give the same treatment to:

- `extensionKitExtensionDependencies` 
- `appClipDependencies`

Reply:
I have added versions with platform conditions for the mentioned methods